modules/gdpr_tasks/gdpr_tasks.module

```diff
@@ -78,185 +75,32 @@ function gdpr_tasks_theme_suggestions_gdpr_task(array $variables) {
-        unset($values['gdpr_rta']);
-        $inc[$key] = $values;
-
-        if ('maybe' === $rta) {
-          $maybe[$key] = $values;
-        }
-      }
-
-      /* @var \Drupal\gdpr_tasks\TaskManager $task_manager */
-      $task_manager = \Drupal::service('gdpr_tasks.manager');
-      $destination = $task_manager->toCsv($inc, $dirname);
-      $export = \file_get_contents($destination);
-
-      // Generate a file entity.
-      // @todo Add headers to csv export.
-      /** @var \Drupal\file\FileInterface $file */
-      $file = _gdpr_tasks_file_save_data($export, $user, $destination, FILE_EXISTS_REPLACE);
-
-      if (FALSE === $file) {
-        drupal_set_message(t('An error occurred while saving the data export file!'), 'error');
-      }
-      else {
-        $values = [
-          'target_id' => $file->id(),
-          'display' => (int) $settings['display_default'],
-          'description' => '',
-        ];
-
-        $gdprTask->sar_export = $values;
-
-        $temp = $task_manager->toCsv($maybe, $dirname);
-        $gdprTask->manual_data = \file_get_contents($temp);
-      }
-
-      break;
+function gdpr_tasks_file_access(FileInterface $file, $operation, AccountInterface $account) {
+  if ('download' === $operation && $file->getOwnerId() === $account->id()) {
+    return AccessResult::allowed();
   }
+
+  return AccessResult::neutral();
 }
 
 /**
- * Saves a file to the specified destination and creates a database entry.
- *
- * @param string $data
- *   A string containing the contents of the file.
- * @param \Drupal\Core\Session\AccountInterface $user
- *   The owner of the file.
- * @param string|null $destination
- *   (optional) A string containing the destination URI. This must be a stream
- *   wrapper URI. If no value or NULL is provided, a randomized name will be
- *   generated and the file will be saved using Drupal's default files scheme,
- *   usually "public://".
- * @param int $replace
- *   (optional) The replace behavior when the destination file already exists.
- *   Possible values include:
- *   - FILE_EXISTS_REPLACE: Replace the existing file. If a managed file with
- *     the destination name exists, then its database entry will be updated. If
- *     no database entry is found, then a new one will be created.
- *   - FILE_EXISTS_RENAME: (default) Append _{incrementing number} until the
- *     filename is unique.
- *   - FILE_EXISTS_ERROR: Do nothing and return FALSE.
+ * Implements hook_entity_type_alter().
  *
- * @return \Drupal\file\FileInterface|false
- *   A file entity, or FALSE on error.
- *
- * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
- * @throws \Drupal\Core\Entity\EntityStorageException
- *
- * @see file_unmanaged_save_data()
+ * Update entity definitions of certain entity types to be excluded from entity
+ * traversal.
  */
-function _gdpr_tasks_file_save_data($data, AccountInterface $user, $destination = NULL, $replace = FILE_EXISTS_RENAME) {
```

Comment:
Why was this function removed? This was a required thing so saving the CSV files works. It was not re-added anywhere, resulting in files not getting saved at all.